google/cloud/sql/connector/InstanceConnectionManager.py

```diff
@@ -38,12 +37,41 @@
 
 
 class InstanceMetadata:
-    ip_addresses: Dict[str, str]
-    ssl_context: OpenSSL.SSL.Context
+    ip_address: str
+    ca_fileobject: NamedTemporaryFile
+    cert_fileobject: NamedTemporaryFile
+    key_fileobject: NamedTemporaryFile
+    context: ssl.SSLContext
 
-    def __init__(self, ssl_context, ip_addresses):
-        self.ssl_context = ssl_context
-        self.ip_addresses = ip_addresses
+    def __init__(
+        self,
+        ephemeral_cert: str,
+        ip_address: str,
+        private_key: str,
+        server_ca_cert: str,
+    ):
+        self.ip_address = ip_address
+
+        self.ca_fileobject = NamedTemporaryFile(suffix=".pem")
+        self.cert_fileobject = NamedTemporaryFile(suffix=".pem")
+        self.key_fileobject = NamedTemporaryFile(suffix=".pem")
+
+        # Write each file and reset to beginning
+        self.ca_fileobject.write(server_ca_cert.encode())
+        self.cert_fileobject.write(ephemeral_cert.encode())
+        self.key_fileobject.write(private_key)
+
+        self.ca_fileobject.seek(0)
+        self.cert_fileobject.seek(0)
+        self.key_fileobject.seek(0)
+
+        self.context = ssl.SSLContext()
```

Comment:
If we are creating the context here, can we make the fileobjects private? They shouldn't be used anywhere else, right?